pkg/platform/model/buildplanner.go

```diff
@@ -101,6 +102,11 @@ func (bp *BuildPlanner) FetchBuildResult(commitID strfmt.UUID, owner, project st
 		if err != nil {
 			return nil, errs.Wrap(err, "failed to poll build plan")
 		}
+
+		build, err = resp.Build()
+		if err != nil {
+			return nil, errs.Wrap(err, "Could not get build from response")
+		}
```

Comment:
I think we need a sleep in between retries here so we don't hammer the buildplan API.

Reply:
The implementations of this function don't make any API calls. They just process the response information and return a `Build` struct.

Reply:
Then what is the thing we're polling? If it's entirely local we should probably be handling this with a channel or a callback.

Reply:
We poll in `pollBuildPlan` which polls every second and returns the response once the build status is no longer `PLANNING`. I've made a slight change to it to return the build so we're not calling the `Build()` function repeatedly. 

The `Build()` function is doing some processing of the graphQL result, checking for errors in the response and returning the build.

Reply:
Oh for some reason my mind was reading this as being in a loop -.- My bad.